pkg/state/state.go

```diff
@@ -7,23 +7,39 @@ import (
 var log = logf.Log.WithName("state")
 
 type RmdNodeData struct {
-	RmdNodeList map[string]string
+	RmdNodeList []string
 }
 
 // NewRmdNodeData() creates an empty RmdNodeData object if no nodestates are present
 // Otherwise, relevant data is extracted from node states and placed in a map
 func NewRmdNodeData() *RmdNodeData {
 	return &RmdNodeData{
-		RmdNodeList: map[string]string{},
+		RmdNodeList: []string{},
 	}
 }
 
 // UpdateRmdNodeData() adds new data to map or updates existing node data
-func (nd *RmdNodeData) UpdateRmdNodeData(nodeName string, namespaceName string) {
-	nd.RmdNodeList[nodeName] = namespaceName
+func (nd *RmdNodeData) UpdateRmdNodeData(nodeName string) {
+	nodeInList := false
+	if len(nd.RmdNodeList) != 0 {
+		for _, node := range nd.RmdNodeList {
+			if nodeName == node {
+				nodeInList = true
```

Comment:
There's a few ways to do this. Simplest is probably to just `return` here if we find the node name. 
Then outside the for loop execute the `append` regardless, it will only ever get to that point if the node doesn't exist. Removes the need for `nodeInList` variable